src/H5SC.c

```diff
@@ -155,21 +157,24 @@ H5SC_flush_dset(H5SC_t *cache, H5D_t *dset, bool evict)
 /*-------------------------------------------------------------------------
  * Function: H5SC_read
  *
- * Purpose:  Reads raw data through a shared chunk cache.
+ * Purpose:  Reads raw data through a shared chunk cache. There may be
+ * datasets in the dset_info array that do not support the shared chunk
+ * cache. These datasets must be ignored by the shared chunk cache. There
+ * may also be datasets that have skip_io set. These datasets must also be
+ * skipped.
```

Comment:
While it may look a bit awkward at the higher levels, I propose that we partition up the `H5D_dset_io_info_t` structs into separate arrays (or do a sort) based on those that support H5SC and those which don't (also the same for the `skip_io` case) to simplify the logic here and not push the concerns of skipping over datasets down to these functions.

Reply:
We could certainly do that. My initial idea is that it won't be difficult or costly to just skip over the unused datasets (as is the case in H5Dio.c, where it skips over datasets that DO support SCC), and is not worth the overhead, but if it ends up being difficult we can create a new list.